utils/staking-miner/src/rpc_helpers.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Helper method for RPC.
+
+use super::*;
+use jsonrpsee_ws_client::traits::Client;
+pub(crate) use jsonrpsee_ws_client::v2::params::JsonRpcParams;
+
+#[macro_export]
+macro_rules! params {
+	($($param:expr),*) => {
+		{
+			let mut __params = vec![];
+			$(
+				__params.push(serde_json::to_value($param).expect("json serialization infallible; qed."));
```

Comment:
Why can expect param serialization to be infallibe?

Reply:
Given your other comments that we want it to fail so issues are clear I think this expect is totally fine